### DISCUSSION TITLE
What is the correct way to use luarocks?

### DISCUSSION THREAD

**Discussion by wushu037** [keplerproject/luarocks#1574] [Q&A] [answered]
I'm having two problems

1. some modules are downloaded with names that don't match the names used by require. E.g. `luarocks install luasocket` -> `require("socket")`
2. does luarocks offer the ability to manage multiple versions of a module? I didn't find a description.

This can be difficult when doing collaborative development and long term development. So:
1. do I need to add a file to the project that describes which modules are used in the project?
2. Since there is no version management, it is possible that everyone in the team installs a different version of the module at a different time, so should I put the module in the project directory?

I need to hear from the luarocks team.

**Comment by alerque** [comment, marked as answer]
You already got [a serviceable answer](https://github.com/lunarmodules/luasocket/issues/406#issuecomment-1666821005) to this question when you asked it on the wrong repository. I don't think you "need" to hear specifically from the LuaRocks term. That is the answer: you can setup a rockspec for your project that specs exactly the versions you need that version to use, then have everybody install those via the appropriate options and optionally into a tree specific to the project.
